source/robot_model/src/Model.cpp

```diff
@@ -334,86 +307,80 @@ Model::inverse_geometry(const state_representation::CartesianState& desired_cart
   Eigen::VectorXd psi(this->robot_model_.nq);
   Eigen::VectorXd err(6);
-  const state_representation::CartesianPose X_d(desired_cartesian_state);
-
-  int i = 0;
-  while (i < params.max_number_of_iterations) {
-    err = ((X_d - this->forward_geometry(q, frame_id)) / dt).data();
-    if (err.cwiseAbs().maxCoeff() < params.tolerance) {
+  for (unsigned int i = 0; i < parameters.max_number_of_iterations; ++i) {
+    err = ((cartesian_pose - this->forward_kinematics(q, frame_id)) / dt).data();
+    // break in case of convergence
+    if (err.cwiseAbs().maxCoeff() < parameters.tolerance) {
       return q;
     }
-
     J = this->compute_jacobian(q);
-    W_b = this->cwln_weighted_matrix(q, params.margin);
+    W_b = this->cwln_weighted_matrix(q, parameters.margin);
     W_c = Eigen::MatrixXd::Identity(this->robot_model_.nq, this->robot_model_.nq) - W_b;
-    psi = params.gamma * this->cwln_repulsive_potential_field(q, params.margin);
+    psi = parameters.gamma * this->cwln_repulsive_potential_field(q, parameters.margin);
     J_b = J * W_b;
     JJt.noalias() = J_b * J_b.transpose();
-    JJt.diagonal().array() += params.damp;
-    dq.set_velocities(W_c * psi
-                          + params.alpha * W_b * (J_b.transpose() * JJt.ldlt().solve(err - J.data() * W_c * psi)));
-    q = q + dq * dt;
+    JJt.diagonal().array() += parameters.damp;
+    dq.set_velocities(W_c * psi + parameters.alpha * W_b * (J_b.transpose() * JJt.ldlt().solve(err - J * W_c * psi)));
+    q += dq * dt;
     q = this->clamp_in_range(q);
-    ++i;
   }
-  throw (exceptions::IKDoesNotConverge(params.max_number_of_iterations, err.array().abs().maxCoeff()));
+  throw (exceptions::InverseKinematicsNotConvergingException(parameters.max_number_of_iterations,
+                                                             err.array().abs().maxCoeff()));
 }
 
 state_representation::JointPositions
-Model::inverse_geometry(const state_representation::CartesianState& desired_cartesian_state,
-                        const std::string& frame_name,
-                        const InverseGeometryParameters& params) {
+Model::inverse_kinematics(const state_representation::CartesianPose& cartesian_pose,
+                          const std::string& frame_name,
+                          const InverseKinematicsParameters& parameters) {
   Eigen::VectorXd q(pinocchio::neutral(this->robot_model_));
-  state_representation::JointPositions pos(this->get_robot_name(), q);
-
-  return this->inverse_geometry(desired_cartesian_state, pos, frame_name, params);
+  state_representation::JointPositions positions(this->get_robot_name(), q);
+  return this->inverse_kinematics(cartesian_pose, positions, frame_name, parameters);
 }
 
-state_representation::CartesianTwist Model::forward_kinematic(const state_representation::JointState& joint_state) {
+state_representation::CartesianTwist Model::forward_velocity(const state_representation::JointState& joint_state) {
```

Comment:
Aren't we missing a `forward_velocity` for a specific frame, or did we say this is not necessary? Would be another PR anyway..

Reply:
indeed that is a good catch. There should be an optional argument as with `forward_kinematics`. I can make it a next PR but it could also go there I think. This is part of refactoring for consistency

Reply:
I don't mind

Reply:
ok let us not bloat this PR which is big enough already. Because there is also potentially the version of the function that also accept a vector of frame names that is missing. I will make a subsequent PR